File: server/internal/dto/request/comment.go
package request

// CreateCommentReq 创建评论请求
type CreateCommentReq struct {
	// TargetType 目标类型：video / article / dynamic
	TargetType string `json:"target_type" binding:"required"`
	// TargetID 目标 ID（视频 ID / 文章 ID / 动态 ID）
	TargetID uint64 `json:"target_id" binding:"required"`
	// ParentID 父评论 ID，0 表示顶级评论
	ParentID uint64 `json:"parent_id"`
	// Content 评论内容，1-2000 字符
	Content string `json:"content" binding:"required,min=1,max=2000"`
}

// ListCommentsReq 评论列表查询请求
type ListCommentsReq struct {
	TargetType string `form:"target_type" binding:"required"`
	TargetID   uint64 `form:"target_id" binding:"required"`
	Page       int    `form:"page"`
	PageSize   int    `form:"page_size"`
}

// ListRepliesReq 回复列表查询请求
type ListRepliesReq struct {
	// TargetType 目标类型，用于路由到对应评论表
	TargetType string `form:"target_type" binding:"required"`
	CommentID  uint64 `form:"comment_id" binding:"required"`
	Page       int    `form:"page"`
	PageSize   int    `form:"page_size"`
}

// CommentLikeReq 评论点赞请求
type CommentLikeReq struct {
	TargetType string `json:"target_type" binding:"required"`
	CommentID  uint64 `json:"comment_id" binding:"required"`
}

// DeleteCommentReq 删除评论请求
type DeleteCommentReq struct {
	TargetType string `json:"target_type" binding:"required"`
	CommentID  uint64 `json:"comment_id" binding:"required"`
}

File: web/src/api/comment.ts
import request from './request'
import type { CommentItem } from '@/types'

// 评论列表（统一接口，通过 target_type 区分 video/article/dynamic）
export function getCommentList(
  target_type: string,
  target_id: number,
  page: number = 1,
  page_size: number = 20
) {
  return request.get<any, { list: CommentItem[]; total: number }>(
    '/comment/list',
    { params: { target_type, target_id, page, page_size } }
  )
}

// 回复列表
export function getCommentReplies(
  target_type: string,
  comment_id: number,
  page: number = 1,
  page_size: number = 20
) {
  return request.get<any, { list: CommentItem[]; total: number }>(
    '/comment/replies',
    { params: { target_type, comment_id, page, page_size } }
  )
}

// 创建评论
export function createComment(
  target_type: string,
  target_id: number,
  content: string,
  parent_id: number = 0
) {
  return request.post<any, { comment_id: number }>('/comment/create', {
    target_type,
    target_id,
    parent_id,
    content,
  })
}

// 点赞评论
export function likeComment(target_type: string, comment_id: number) {
  return request.post<any, void>('/comment/like', { target_type, comment_id })
}

// 取消点赞评论
export function unlikeComment(target_type: string, comment_id: number) {
  return request.post<any, void>('/comment/unlike', { target_type, comment_id })
}

// 删除评论
export function deleteComment(target_type: string, comment_id: number) {
  return request.post<any, void>('/comment/delete', { target_type, comment_id })
}
